skorch/net.py

```diff
@@ -1377,6 +1377,14 @@ def check_is_fitted(self, attributes=None, *args, **kwargs):
         )
         check_is_fitted(self, attributes, *args, **kwargs)
 
+    def __sklearn_is_fitted__(self):
+        """This method is called when sklearn's ``check_is_fitted`` is used.
+        
+        Explained here: 
+        https://scikit-learn.org/stable/auto_examples/developing_estimators/sklearn_is_fitted.html
+        """
+        return self.check_is_fitted()
```

Comment:
Note that `self.check_is_fitted()` does not return a bool, instead it raises a `NotFittedError`. So for the signature of this method to be correct, we'd have to call `self.check_is_fitted()`, intercept the error, and return based on whether the error was raised or not.

The tests still pass, as they check for the very error that is raised here, but I don't think that's the intent. The error, and the message, should correspond to what sklearn would return, since we explicitly call `check_is_fitted` from sklearn, i.e. the message should be something like:

> `"This NeuralNetBinaryClassifier instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator."`